runtime_builders/builder_util.py

```diff
@@ -57,10 +64,64 @@ def get_file_from_gcs(gcs_file, temp_file):
         return False
 
 
-def write_manifest_file(manifest):
-    manifest_contents = yaml.round_trip_dump(manifest,
-                                             default_flow_style=False)
-    write_to_gcs(MANIFEST_FILE, SCHEMA_LINE + manifest_contents)
+def verify_manifest(manifest):
+    """Verify that the provided runtime manifest is valid before publishing.
+
+    Aliases are provided for runtime 'names' that can be included in users'
+    application configuration files: this method ensures that all the aliases
+    can resolve to actual builder files.
+
+    All builders and aliases are turned into nodes in a graph, which is then
+    traversed to be sure that all nodes lead down to a builder node.
+
+    Example formatting of the manifest, showing both an 'alias' and
+    an actual builder file:
+
+    runtimes:
+      java:
+        target:
+          runtime: java-openjdk
+      java-openjdk:
+        target:
+          file: gs://runtimes/java-openjdk-1234.yaml
+        deprecation:
+          message: "openjdk is deprecated."
+    """
+    try:
+        node_graph = {}
+        for key, val in manifest.get('runtimes').iteritems():
+            target = val.get('target', {})
+            if not target:
+                deprecation = val.get('deprecation', {})
+                if not deprecation:
+                    logging.error('No target or deprecation specified for '
+                                  'runtime: {0}'.format(key))
+                    sys.exit(1)
+                continue
+            child = None
+            isBuilder = 'file' in target.keys()
+            if not isBuilder:
+                child = target['runtime']
+            node = node_graph.get(key, {})
+            if not node:
+                node_graph[key] = Node(key, isBuilder, child)
+        for _, node in node_graph.items():
+            child = node
+            while True:
```

Comment:
Hmm, well one thing I should definitely do is at least keep track of what I've seen, so if there's a circular dependency this doesn't infinite loop